internal/health/server.go

```diff
@@ -7,27 +7,49 @@ import (
 	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 // Server has methods implementing of server rpc.
 type Server struct {
 	core *Core
 }
 
+// ReadinessCheck checks only for the responsiveness of the service
+func (h *Server) ReadinessCheck(ctx context.Context, _ *emptypb.Empty) (*metrov1.StatusCheckResponse, error) {
+	if !h.core.IsHealthy(ctx) {
+		logger.Ctx(ctx).Debugw("metro health check", "status", "Unhealthy")
+		return nil, status.Error(codes.Unavailable, "Unhealthy")
+	}
+
+	logger.Ctx(ctx).Debugw("metro readiness check", "status", "Healthy")
+	return &metrov1.StatusCheckResponse{
+		ServingStatus: metrov1.StatusCheckResponse_SERVING_STATUS_SERVING,
+	}, nil
+}
+
+// LivenessCheck checks responsiveness of all the dependant resources that the service is using
+func (h *Server) LivenessCheck(ctx context.Context, _ *emptypb.Empty) (*metrov1.StatusCheckResponse, error) {
```

Comment:
by default all grpc apis are being tracked under opentracing, we can remove liveliness prove from there as k8s checks that every sec and adds to lot of volume

Reply:
ah yes! let me filter these out.

Reply:
anyways will make this change in a different PR. Need to figure out the middleware interceptor.

Reply:
Sure, Works for now

Reply:
Tracking here https://github.com/razorpay/metro/issues/183